source/elements/oneMKL/source/domains/blas/gemv_batch.rst

```diff
@@ -0,0 +1,579 @@
+   }
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::row_major {
+       void gemv_batch(sycl::queue &queue,
+                       onemkl::transpose trans,
+                       std::int64_t m,
+                       std::int64_t n,
+                       T alpha,
+                       sycl::buffer<T,1> &a,
+                       std::int64_t lda,
+                       std::int64_t stridea,
+                       sycl::buffer<T,1> &x,
+                       std::int64_t incx,
+                       std::int64_t stridex,
+                       T beta,
+                       sycl::buffer<T,1> &y,
+                       std::int64_t incy,
+                       std::int64_t stridey,
+                       std::int64_t batch_size)
+   }
+
+.. container:: section
+
+   .. rubric:: Input Parameters
+
+   queue
+      The queue where the routine should be executed.
+
+   trans
+      Specifies op(``A``) the transposition operation applied to the
+      matrices ``A``. See :ref:`onemkl_datatypes` for more details.
+
+   m
+      Number of rows of op(``A``). Must be at least zero.
+
+   n
+      Number of columns of op(``A``). Must be at least zero.
+
+   alpha
+      Scaling factor for the matrix-matrix products.
+
+   a
+      Buffer holding the input matrices ``A`` with size ``stridea`` * ``batch_size``.
+
+   lda
+      The leading dimension of the matrices ``A``. It must be positive.
+
+      .. list-table::
+         :header-rows: 1
+
+         * -
+           - ``A`` not transposed
+           - ``A`` transposed
+         * - Column major
+           - ``lda`` must be at least ``m``.
+           - ``lda`` must be at least ``k``.
+         * - Row major
+           - ``lda`` must be at least ``k``.
+           - ``lda`` must be at least ``m``.
+
+   stridea
+      Stride between different ``A`` matrices.
+
+   x
+      Buffer holding the input matrices ``X`` with size ``stridex`` * ``batch_size``.
+
+   incx
+      The stride of the vector ``X``. It must be positive.
+
+   stridex
+      Stride between different consecutive ``X`` matrices, must be at least 0.
+
+   beta
+      Scaling factor for the vector ``Y``.
+
+   y
+      Buffer holding input/output vectors ``Y`` with size ``stridey`` * ``batch_size``.
+
+   incy
+      Stride between two consecuctive elements of the ``y`` vectors.
+
+   stridey
+      Stride between two consecutive ``Y`` vectors. Must be at least
+      (1 + (len-1)*abs(incy)) where ``len`` is ``m`` if the matrix ``A``
+      is non transpose or ``n`` otherwise.
+
+   batch_size
+      Specifies the number of matrix-vector operations to perform.
+
+.. container:: section
+
+   .. rubric:: Output Parameters
+
+   y
+      Output overwritten by ``batch_size`` matrix-vector product
+      operations of the form ``alpha`` * op(``A``) * ``X`` + ``beta`` * ``Y``.
+
+.. container:: section
+
+   .. rubric:: Notes
+
+   If ``beta`` = 0, vector ``Y`` does not need to be initialized before
```

Comment:
I am not sure if this is supported for gemv. since we dont have it in gemv.rst, i would remove it